### ISSUE TITLE
Cannot run train due to flash

### ISSUE THREAD

**Issue by david-waterworth** [karpathy/nanoGPT#104]
I noticed the comment that you're using torch 2.0 and if you encounter warnings to set `--compile=False`

Problem I'm running into is flash is auto-detected

        # flash attention make GPU go brrrrr but support is only in PyTorch nightly and still a bit scary
        self.flash = hasattr(torch.nn.functional, 'scaled_dot_product_attention')
        if not self.flash:
            print("WARNING: using slow attention, install PyTorch nightly for fast Flash Attention")
            # causal mask to ensure that attention is only applied to the left in the input sequence
            self.register_buffer("bias", torch.tril(torch.ones(config.block_size, config.block_size))
                                        .view(1, 1, config.block_size, config.block_size))

But then dropout must be zero, i.e. model.py#68

        # causal self-attention; Self-attend: (B, nh, T, hs) x (B, nh, hs, T) -> (B, nh, T, T)
        if self.flash:
            # efficient attention using Flash Attention CUDA kernels
            assert self.dropout == 0.0, "need dropout=0.0 for now, PyTorch team is working on fix in #92917"
            y = torch.nn.functional.scaled_dot_product_attention(q, k, v, attn_mask=None, dropout_p=self.dropout, is_causal=True)
        else:
            # manual implementation of attention

I set `dropout==0.0` in the toy example as a check and it worked, but it seems that it's probably better to disable `flash` rather than dropout until it is supported by flash?

**Comment by david-waterworth** [comment]
PS Does `torch.nn.functional.scaled_dot_product_attention` actually implement FlashAttention? I get the impression reading https://github.com/pytorch/pytorch/pull/92917 that its not merged yet? I got a warning when I disabled the assert (i.e. used dropout!=0.0) but it did appear to work.

**Comment by karpathy** [comment]
oh, damn, i think you're right sorry about this... yes let me fix quickly

**Comment by karpathy** [comment]
ok fixed, this was dumb on my part thank you for flagging the issue. closing.

**Comment by dinakar17** [comment]
!python prepare.py
!python train.py config/finetune_shakespeare.py


Encountered the following issue in my console 

```
WARNING: using slow attention. Flash Attention atm needs PyTorch nightly and dropout=0.0
WARNING: using slow attention. Flash Attention atm needs PyTorch nightly and dropout=0.0
WARNING: using slow attention. Flash Attention atm needs PyTorch nightly and dropout=0.0
number of parameters: 1555.97M
^C
````
What should be done for training to run properly?
